apps/web/src/modules/create-proposal/components/TransactionForm/Escrow/EscrowForm.schema.ts

```diff
@@ -59,21 +59,35 @@ export const MilestoneSchema = yup.object({
   description: yup.string(),
 })
 
-export const EscrowFormSchema = yup.object({
-  clientAddress: addressValidationSchema,
-  recipientAddress: addressValidationSchema,
-  safetyValveDate: yup
-    .date()
-    .required('Safety valve date is required.')
-    .min(
-      new Date(Date.now() + 30 * 24 * 60 * 60 * 1000),
-      'Safety valve date must be at least 30 days from today.'
+export const EscrowFormSchema = yup
+  .object({
+    clientAddress: addressValidationSchema,
+    recipientAddress: addressValidationSchema.test(
+      'not-same-as-client',
+      'Recipient address must be different from client address',
+      function (value) {
+        return value !== this.parent.clientAddress
+      }
     ),
-  milestones: yup
-    .array()
-    .of(MilestoneSchema)
-    .min(1, 'At least one milestone is required'),
-})
+    safetyValveDate: yup
+      .date()
+      .required('Safety valve date is required.')
+      .min(
+        new Date(Date.now() + 30 * 24 * 60 * 60 * 1000),
+        'Safety valve date must be at least 30 days from today.'
```

Comment:
if possible can we check if safety valve date is at least 30 days from the last end date of the milestones?

Reply:
done!!